File: final-prices-with-a-special-discount-in-a-shop.ts
{
    const finalPrices = function(prices) {
        const result = [];

        loop: for(let i = 0; i < prices.length; i++) {
            for(let j = i + 1; j < prices.length; j++) {
                if(prices[i] >= prices[j]) {
                    result.push(prices[i] - prices[j])
                    continue loop;
                }
            }

            result.push(prices[i])
        }

        return result
    };

    console.log(finalPrices([8,4,6,2,3]))
}

File: final-prices-with-a-special-discount-in-a-shop.rs
pub fn final_prices(prices: Vec<i32>) -> Vec<i32> {
    let mut result = Vec::new();

    'main: for i in 0..prices.len() {

        for j in i + 1..prices.len() {
            if prices[i] >= prices[j] {
                result.push(prices[i] - prices[j]);
                continue 'main;
            }
        }

        result.push(prices[i]);
    }

    return  result;
}